src/tests/unittests/hextobin.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * hextobin.c: Test suite for sc_hex_to_bin()
+ *
+ * Copyright (C) 2022 Peter Popovec <popovec.peter@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+#include "libopensc/opensc.h"
+
+#define LEN 15
+
+int main()
+{
+	char *t;
+	char test[][LEN] = {
+/* result length (0x60 error string, 0x70 end of array), string, result string */
+
+		{1}, "0", "\x00",
+		{1}, ":0", "\x00",
+		{1}, " :0 :", "\x00",
+		{1}, "00", "\x00",
+		{1}, ":00", "\x00",
+		{1}, "0:0 :", "\x00",
+
+		{1}, "1", "\x01",
+		{1}, " 9", "\x09",
+		{1}, "d", "\x0d",
+		{1}, ":a", "\x0a",
+		{1}, " :b :", "\x0b",
+		{1}, " :c ", "\x0c",
+
+		{1}, "01", "\x01",
+		{1}, " 09", "\x09",
+		{1}, ":0a", "\x0a",
+		{1}, " :0b :", "\x0b",
+
+		{1}, "10", "\x10",
+		{1}, " 90", "\x90",
+		{1}, ":a0", "\xa0",
+		{1}, " :B0 :", "\xb0",
+
+		{1}, " 11:", "\x11",
+		{1}, " :11:", "\x11",
+		{1}, " :1 1:", "\x11",
+		{1}, "1:1", "\x11",
```

Comment:
I am wondering if this is the expected output. I would expect `\x01\x01`, but it might be a matter of convention/taste/mood.